packages/scoutgame/src/points/calculateBuilderLevel.ts

```diff
@@ -81,38 +94,30 @@ export async function calculateBuilderLevels({
       }
 
       if (!acc[builderId]) {
+        // Find index of first active week in all season weeks
+        const firstActiveWeekIndex = weeksWindow.indexOf(gemsPayout.week);
+
+        // Get number of weeks builder has been active (from first week to end of season)
+        const activeWeeks = weeksWindow.slice(firstActiveWeekIndex).length;
         acc[builderId] = {
           builderId,
           totalPoints: 0,
           firstActiveWeek: gemsPayout.week,
+          activeWeeks,
           centile: 0,
           level: 0,
           averageGemsPerWeek: 0
         };
       }
+
       acc[builderId].totalPoints += gemsPayout.points;
+      acc[builderId].averageGemsPerWeek = Math.floor(acc[builderId].totalPoints / acc[builderId].activeWeeks);
```

Comment:
instead of looping thru builders again, just calculate the avg again each time we find another week with points